src/main/java/uk/gov/justice/hmpps/datacompliance/services/DeletionReferralService.java

```diff
@@ -14,8 +17,26 @@
 
     private TimeSource timeSource;
     private OffenderDeletionReferralRepository offenderDeletionReferralRepository;
+    private OffenderDeletionGrantedEventPusher offenderDeletionGrantedEventPusher;
 
-    public void storeOffenderDeletionReferral(final OffenderPendingDeletionEvent event) {
+    public void handlePendingDeletion(final OffenderPendingDeletionEvent event) {
+        storeOffenderDeletionReferral(event);
+
+        // TODO GDPR-51 Complete retention checks on offender record
+
+        offenderDeletionGrantedEventPusher.sendEvent(event.getOffenderIdDisplay());
```

Comment:
no danger of this actually triggering detenions of stuff?

Reply:
It's not deployed to production, and there's nothing kicking the whole process off so pretty safe